cg/apps/housekeeper/hk.py

```diff
@@ -649,3 +653,13 @@ def update_archive_retrieved_at(
         for archive in archives_to_update:
             archive.retrieval_task_id = new_retrieval_job_id
         self._store.session.commit()
+
+    def get_spring_files_retrieved_before(self, date: dt.datetime):
+        return self._store.get_files_retrieved_before(date, tag_names=[SequencingFileTag.SPRING])
+
+    def reset_retrieved_archive_data(self, files_to_reset: list[File]):
+        """Resets 'retrieval task id' and 'retrieved at' for all files' corresponding archive entries"""
+        for file in files_to_reset:
+            file.archive.retrieval_task_id = None
+            file.archive.retrieved_at = None
```

Comment:
Is there a way that we can keep track of retrievals? Maybe we can just reset the retrieval_task_id, but kee the retrieved_at date?

Reply:
Currently we don't really see a need for it, so we have decided to not implement a way to track it. Open to do it if there is a need, but I would rather we not have the `retrieved_at` set for files not on Hasta, than to keep track of the latest retrieval this way.